src/chunk.c

```
@@ -3372,17 +3400,35 @@ ts_chunk_set_unordered(Chunk *chunk)
 	return ts_chunk_add_status(chunk, CHUNK_STATUS_COMPRESSED_UNORDERED);
 }
 
+/*No inserts,updates and deletes are permitted on a frozen chunk.
+ * Compression policies etc do not run on a frozen chunk.
+ * Only valid operation is dropping the chunk
+ */
 bool
-ts_chunk_add_status(Chunk *chunk, int32 status)
+ts_chunk_set_frozen(Chunk *chunk)
 {
-	return ts_chunk_set_status(chunk, ts_set_flags_32(chunk->fd.status, status));
+#if PG14_GE
+	return ts_chunk_add_status(chunk, CHUNK_STATUS_FROZEN);
+#else
+	return false;
```

This should error instead of failing silently.